### DISCUSSION TITLE
Solution diverges while using Axisymmetric

### DISCUSSION THREAD

**Discussion by mohamedsemz** [su2code/SU2#1291] [Q&A]
I'm trying to run simulation on delaval nozzle 2D. and using Axisymmetric. I want to define the Pressure and temperature at the inlet, pressure at outlet and symmetry on the rotation axis.

So I used the following configuration and the solution diverges after ** iteration with error: "SU2 has diverged. (NaN detected)"
`
% -------------------- COMPRESSIBLE FREE-STREAM DEFINITION --------------------%

% Mach number (non-dimensional, based on the free-stream values)
MACH_NUMBER= 1e-9

% Angle of attack (degrees, only for compressible flows)
AOA= 0.0

% Side-slip angle (degrees, only for compressible flows)
SIDESLIP_ANGLE= 0.0

% Init option to choose between Reynolds (default) or thermodynamics quantities
% for initializing the solution (REYNOLDS, TD_CONDITIONS)
INIT_OPTION= REYNOLDS

% Free-stream option to choose between density and temperature (default) for
% initializing the solution (TEMPERATURE_FS, DENSITY_FS)
FREESTREAM_OPTION= TEMPERATURE_FS

% Free-stream pressure (101325.0 N/m^2, 2116.216 psf by default)
FREESTREAM_PRESSURE= 11e5

% Free-stream temperature (288.15 K, 518.67 R by default)
FREESTREAM_TEMPERATURE= 303.0

% -------------------- BOUNDARY CONDITION DEFINITION --------------------------%
MARKER_EULER= (Nozzle)
MARKER_SYM = (Symmetry)
MARKER_INLET= ( Inlet, 303.0, 11e5, 1.0, 0.0, 0.0 )
MARKER_OUTLET= (Outlet, 1e5)
`

So what is the possible issues ?

**Comment by bigfooted** [comment]
Hi,
I looked into the axisymmetric convergence for incompressible flows last week and I found the same problem. The issue there was with the viscous terms [of the axisymmetry source term] for the momentum equations. A solution was to add the Jacobian term of the [axisymmetry] source term contribution. 
Are you using the compressible flow solver with a Mach number of 1e-9?
Do you get a converging solution when you switch axisymmetric off?

**Reply by mohamedsemz** [reply]
thank you for your reply.

I’m sorry I didn’t understand the solution can show me how to fix it ?

As I understand, the mach number defined here is the mach at inlet but the expected mach at exit is 2.2

I checked it now and no I don’t get solution also when it’s off

**Reply by mohamedsemz** [reply]
Do you think I have wrong boundary condition definitions ?

What I intend to do is having Po = 11 bar and inlet with very small speed and atmospheric pressure at exit.

Is this boundary conditions defined in configuration matches what I want ?

**Reply by bigfooted** [reply]
ok, then your issue is not related to the axisymmetric source terms. I do not understand now how your boundary conditions are set up.  Do you have something similar to this tutorial case (without the special nonideal gas):
https://su2code.github.io/tutorials/NICFD_nozzle/

**Reply by mohamedsemz** [reply]
I will try it with my case without axisymmetry and then get back to you with the results

**Reply by mohamedsemz** [reply]
Yes I tried it on my case and it works fine now with axisymmetric is either enabled or disabled.
I have another question, How to calculate forces acting on the nozzle wall ?

Thank you.
